packages/sanity/src/desk/panes/documentList/listenSearchQuery.ts

```diff
@@ -69,12 +68,11 @@ export function listenSearchQuery(options: ListenQueryOptions): Observable<Sanit
     share(),
   )
 
-  const [welcome$, mutationAndReconnect$] = partition(events$, (ev) => ev.type === 'welcome')
-
   return merge(
-    welcome$.pipe(take(1)),
-    mutationAndReconnect$.pipe(throttleTime(1000, asyncScheduler, {leading: true, trailing: true})),
+    of({type: 'welcome', visibility: null}),
+    events$.pipe(filterEvents((ev) => ev.type !== 'welcome')),
   ).pipe(
+    throttleTime(1000, asyncScheduler, {leading: true, trailing: true}),
```

Comment:
I think the current logic addresses an edge case that rarely happens, so I'm not against merging this in it's current shape.

> Are there any unforeseen consequences with this approach?

I find it hard to say with confidence - it will _rarely_ be an issue for sure, but what downstream effects there are _when_ it happens is hard to say. I'm not sure how much third parties rely on this function and the current guarantees it provides.

> Do document lists still function as well as before, in terms of throttling, loading etc?

Yes, the logic seems fine and it worked as intended when I tested locally

> Do you agree with the trade-off described?

Given that it's such an edge case, I think it's fair, but I'd expect it to occationally display a slightly outdated list for a tiny percentage of studio users, which is probably ok.

> Are there alternative solutions to this that we could implement that I have not considered?

Not sure if you considered, but if we instead make the trade-off be _an extra initial request_, we can keep the current guarantees by doing this:
```suggestion
  const [welcome$, mutationAndReconnect$] = partition(events$, (ev) => ev.type === 'welcome')

  return merge(
    of({type: 'init', visibility: null}),
    welcome$.pipe(take(1)),
    mutationAndReconnect$.pipe(throttleTime(1000, asyncScheduler, {leading: true, trailing: true})),
  ).pipe(
```